Sources/Inject/Integrations/SwiftUI.swift

```diff
@@ -2,30 +2,35 @@ import Foundation
 import SwiftUI
 
 #if DEBUG
-public extension SwiftUI.View {
-    func enableInjection() -> some SwiftUI.View {
+@propertyWrapper
+public struct Injection: DynamicProperty {
+    @ObservedObject private var iO = Inject.observer
+    public init(animation: Animation? = nil) {
         _ = Inject.load
-        
-        // Use AnyView in case the underlying view structure changes during injection.
-        // This is only in effect in debug builds.
-        return AnyView(self)
+        Inject.animation = animation
     }
-
-    func onInjection(callback: @escaping (Self) -> Void) -> some SwiftUI.View {
-        onReceive(Inject.observer.objectWillChange, perform: {
+    public private(set) var wrappedValue: Inject.Type = Inject.self
+}
+public extension SwiftUI.View {
+    func enableInjection() -> some SwiftUI.View { AnyView(self) }
+    func enableInjection(callback: @escaping (Self) -> Void) -> some SwiftUI.View {
+        AnyView(self.onReceive(Inject.observer.objectWillChange, perform: {
             callback(self)
-        })
-        .enableInjection()
+        }))
     }
 }
 
 #else
 public extension SwiftUI.View {
+    @propertyWrapper
```

Comment:
This doesn't seem to be valid Swift. I get the error: `type 'Injection' cannot be nested in protocol extension of 'View'`

Reply:
Sorry not sure when that happened, line 24 belonged at line 29! Release build working with that fixed